### ISSUE TITLE
Tests: core_tests caching generation from the test perspective

### ISSUE THREAD

**Issue by mj-xmr** [monero-project/monero#7442]
I still want to speed up the core tests, by addressing the generation bottleneck [mentioned here](https://github.com/monero-project/monero/pull/7300), but without affecting the implementation itself. 

For a recap, I discovered there, that 30% of the core_tests time is taken by the generation step, so I wanted to cache it by diving deep into the implementation and putting there a cache. The solution was not accepted for valid reasons. I would like to propose a different solution: give the core_tests a new parameter: `--generate-cache-and-play`, which would perform serialization on the first run, and deserialization on the second run. For safety, the cache would have an ID, based on the md5 sum of the core_tests binary.

Would such a solution be accepted? Any objections?

**Comment by moneromooo-monero** [comment]
Maybe you should consider using --generate_test_data and --play_test_data instead (and fix if broken).

**Comment by mj-xmr** [comment]
Yes, it makes sense. It was somehow broken, although not beyond repair. I'm on it.

**Comment by analyticname857** [comment]
Have a look at #7518

**Comment by hardrun744** [comment]
View #7521

**Comment by velvetbailiff** [comment]
Take a look #7523

**Comment by SneakyPlant43** [comment]
#7527

**Comment by LenientVedism280** [comment]
Take a look #7529

**Comment by wideFaint903** [comment]
Take a look #7532

**Comment by finelyscuff** [comment]
View #7534

**Comment by YetCord** [comment]
Take a look #7536

**Comment by Morbidlystrum671** [comment]
View #7540

**Comment by flushRelay34** [comment]
Check out #7543

**Comment by Smartlycount707** [comment]
Check out #7547

**Comment by Lastsew987** [comment]
View #7558

**Comment by Wiselyconserve14** [comment]
Have a look at #7564

**Comment by LuridlyPearl** [comment]
See #7567

**Comment by PlopDelegate54** [comment]
View #7569

**Comment by shiftilyConnect55** [comment]
#7574

**Comment by publiclySpring87** [comment]
#7581

**Comment by potentlywhistle** [comment]
Take a look #7583

**Comment by AnglicanLordship** [comment]
See #7587

**Comment by Truehappen73** [comment]
Check out #7595

**Comment by czechforth** [comment]
See #7597

**Comment by SicilianAsian** [comment]
Take a look #7599

**Comment by civilsalsify58** [comment]
#7601

**Comment by SavagelyExtract49** [comment]
Check out #7603

**Comment by turgidlyforedoom** [comment]
Have a look at #7607

**Comment by selsta** [comment]
We solved slow core tests on CI with #10686.
